samples/cpp/image_generation/text2image_concurrency.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2023-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+#include <iostream>
+
+#include "openvino/genai/image_generation/text2image_pipeline.hpp"
+
+
+#include "imwrite.hpp"
+#include "progress_bar.hpp"
+
+int32_t main(int32_t argc, char* argv[]) try {
+    OPENVINO_ASSERT(argc >= 3, "Usage: ", argv[0], " <MODEL_DIR> '<PROMPT>' '<PROMPT>' ...");
+
+    const std::string models_path = argv[1];
+    const std::string device = "CPU";  // GPU can be used as well
+
+    std::vector<std::string> prompts;
+    for (int i = 2; i < argc; ++i) {
+        prompts.push_back(argv[i]);
+    }
+
+    ov::genai::Text2ImagePipeline pipe(models_path, device);
+
+    std::vector<std::thread> threads;
+
+    for (size_t i = 0; i < prompts.size(); ++i) {
+        threads.emplace_back([i, &pipe, models_path, prompts] () {
+            std::cout << "Starting to generate with prompt: '" << prompts[i] << "'..." << std::endl;
+
+            ov::genai::Text2ImagePipeline request = pipe.clone();
```

Comment:
As far as I understand we should clone pipeline outside of thread

Reply:
Why is that? This exactly mimics the use case inside OpenVINO Model Server.

In OVMS, we **do not** clone pipelines before hand - we **do** it in separate thread when HTTP generaton request appears. We keep the original compiled pipeline in memory, and do not use it for other tasks than being the clone source.

However, this is only a sample. Both scenarios should work, but I can change this sample to work differently. What do you think? @likholat

Reply:
I think `ireq.get_compiled_mode().create_infer_request()` isn't thread safe. Am I wrong?

Reply:
In OpenVINO there are `create_infer_request` stress tests which run creation in separate threads: https://github.com/openvinotoolkit/openvino/blob/master/tests/stress_tests/unittests/tests.cpp#L40-L42
So I'm assuming this is thread safe.

Reply:
It doesn't look like the same case, it creates new infer request each time from a new separate model object:

https://github.com/openvinotoolkit/openvino/blob/9315206a0eed9605fb96a82b8c3ba8686c0c616a/tests/stress_tests/unittests/tests_pipelines/tests_pipelines.cpp#L61

https://github.com/openvinotoolkit/openvino/blob/9315206a0eed9605fb96a82b8c3ba8686c0c616a/tests/stress_tests/common/ie_pipelines/pipelines.cpp#L66